### ISSUE TITLE
Allow comment edits within an admin-specified time period

### ISSUE THREAD

**Issue by mikeschinkel** [calmPress/calmpress#170]
As a user one of the worst things is posting a comment and immediately seeing a typo that you cannot fix.  Sites that allow comments to be edited for a period of time afterward are a godsend for users who commit more typos than they care to admit.

**Comment by nylen** [comment]
> Sites that allow comments to be edited for a period of time afterward to a godsend for users

`s/to a/are a/` :wink: 

+1, good suggestion.

**Comment by markkap** [comment]
The amount of times I wanted sites to have something like that.... the easiest way is probably to not publish immediately to let the user chance to edit it, or just delete it. Once published emails might get sent, comments cached in page cache etc.

**Comment by stefanos82** [comment]
Why not adapt the GitHub concept? When you edit a comment inside a ticket, it keeps comment history so you or anyone else can see the original message after the editing.

If you delete it, it obviously erases the comment's history as well.

I like this kind of concept.

**Comment by mikeschinkel** [comment]
@stefanos82 That sounds like a great idea. So effectively adding revisions, but for comments _(which means we'd need "[comment types](https://petitions.classicpress.net/posts/90/implement-custom-comment-types-and-status).")_

**Comment by nylen** [comment]
cc @dshanske

**Comment by markkap** [comment]
yes, great idea @stefanos82. I am not sure that people will want something like that out of the box and since it is front end related I am afraid of imposing too much additional styling on theme authors, so maybe this is more of a "core plugin" territory.

**Comment by stefanos82** [comment]
Well, we could add it as `mu-plugins/extending-comments.php` and improve its existing mechanism. Those who don't like this feature, they can simply either comment out the file or delete it; it should come bundled with `calmPress` by default though.

Another option would be to implement it as a core plugin that could be enabled by default and users could or should be able to disable it via `Settings > Discussion`.
